### PR TITLE
feat: Enforce asset tagging before switching assets

### PR DESCRIPTION
**Pull request by wbreza**
Enforces that there are no untagged regions within an asset before navigating away from the current loaded asset.  Displays an alert warning when user attempt to navigate.

image

Resolved [AB#18070](https://dev.azure.com/dwrdev/web/wi.aspx?pcguid=80f9922c-b416-454c-92c6-ab7b6867e0b2&id=18070)

### CODE REVIEW

File: src/react/components/common/assetPreview/videoAsset.test.tsx [hunk 1]
```diff
@@ -107,124 +127,133 @@ describe("Video Asset Component", () => {
...
-    });
+            expect(videoPlayerMock.prototype.pause).toBeCalled();
+            expect(videoPlayerMock.prototype.seek).toBeCalledWith(expectedAsset.timestamp);
+            expect(onChildSelectedHandler).toBeCalledWith(expectedAsset);
+        });
 
-    it("moves to the next expected frame when clicking the next button", () => {
-        const childAssets = MockFactory.createChildVideoAssets(defaultProps.asset);
-        const currentAsset = childAssets[0];
+        it("blockings moving to the previous tagged frame when navigation is cancelled", () => {
```
Comment (mydiemho) [inline src/react/components/common/assetPreview/videoAsset.test.tsx]: Nit: I would put the navigation cancelled tests inside another describe

File: src/react/components/pages/editorPage/editorPage.test.tsx [hunk 2]
```diff
@@ -191,13 +194,105 @@ describe("Editor Page Component", () => {
...
+        expect(saveAssetMetadataSpy).not.toBeCalled();
+        expect(saveProjectSpy).not.toBeCalled();
+
+        // Apply tag to region
+        newRegion.tags = ["test"];
+        wrapper.find(Canvas).props().onAssetMetadataChanged(assetMetadata);
+
+        await MockFactory.flushUi();
+
+        expect(editorPage.state().isValid).toBe(true);
```
Comment (mydiemho) [inline src/react/components/pages/editorPage/editorPage.test.tsx]: the test description is about testing invalid state but the test seems to be testing valid state

File: src/react/components/pages/editorPage/editorPage.tsx [hunk 3]
```diff
@@ -73,6 +72,10 @@ export interface IEditorPageState {
     lockedTags: string[];
     /** Size of the asset thumbnails to display in the side bar */
     thumbnailSize: ISize;
+    /** Whether or not the editor is in a valid state */
```
Comment (mydiemho) [inline src/react/components/pages/editorPage/editorPage.tsx]: I would explain what is consider valid

File: src/react/components/pages/editorPage/editorPage.test.tsx (hunk 2 above)
Comment (wbreza) [reply src/react/components/pages/editorPage/editorPage.test.tsx]: If you look through the whole test it tests both conditions.  The page goes into an invalid state, gets fixed and verifies it ends up in a valid state

File: src/react/components/common/assetPreview/videoAsset.test.tsx (hunk 1 above)
Comment (wbreza) [reply src/react/components/common/assetPreview/videoAsset.test.tsx]: They are all about navigating to and from assets - so I will keep them here for now.

File: src/common/localization/en-us.ts [hunk 4]
```diff
@@ -256,6 +256,13 @@ export const english: IAppStrings = {
                 confirmation: "Are you sure you want to remove all regions?",
             },
         },
+        messages: {
+            enforceTaggedRegions: {
+                title: "Invalid region(s) detected",
+                // tslint:disable-next-line:max-line-length
+                description: "1 or more regions have not been tagged.  Please tag all regions before continuing to next asset.",
```
[the remark this answers is not in the document]
Comment (wbreza) [reply src/common/localization/en-us.ts]: Changed to `1 or more regions have not been tagged.  Ensuare all regions are tagged before continuing to next asset.`

File: src/common/localization/en-us.ts [hunk 5]
```diff
@@ -260,7 +260,7 @@ export const english: IAppStrings = {
             enforceTaggedRegions: {
                 title: "Invalid region(s) detected",
                 // tslint:disable-next-line:max-line-length
-                description: "1 or more regions have not been tagged.  Please tag all regions before continuing to next asset.",
+                description: "1 or more regions have not been tagged.  Ensuare all regions are tagged before continuing to next asset.",
```
Comment (mydiemho) [inline src/common/localization/en-us.ts]: Ensuare/Ensure

Comment (wbreza) [reply src/common/localization/en-us.ts]: Fixed

**Pull request merged**